packages/core/admin/admin/src/content-manager/hooks/useDragAndDrop.js

```diff
@@ -46,49 +46,112 @@ export const useDragAndDrop = (
 ) => {
   const objectRef = useRef(null);
 
-  const [{ handlerId }, dropRef] = useDrop({
+  const [{ handlerId, isOver }, dropRef] = useDrop({
     accept: type,
     collect(monitor) {
       return {
         handlerId: monitor.getHandlerId(),
+        isOver: monitor.isOver(),
       };
     },
+    drop(item) {
+      const draggedItem = item.index;
+      const newIndex = index;
+
+      if (isOver && onDropItem) {
+        onDropItem(draggedItem, newIndex);
```

Comment:
How come you've needed to specifically add an onDrop callback? what was the use case? At the moment we just move things and when you let go, you let go? 🤔

Reply:
The previous experience wasn't the experience we wanted. It moved things around as we were still dragging, which in our case was confusing, instead we want to have the same UX as Notion where we just show a line where it a block would land without moving anything before the drop. And before if you dragged a block from the bottom to the top you'd have to cmd+Z many times to undo it, now it's just 1 action to rollback

Reply:
ahhhhhhhhhhhhh nice 👍🏼